pkg/cache/metrics.go

```diff
@@ -32,10 +37,10 @@ type MetricsCollector struct {
 }
 
 func init() {
-	// Initialize the singleton MetricsCollector.
+	// Initialize the singleton BaseCollector.
 	// Set up Prometheus counters for cache operations (hits, misses, sets, deletes, clears).
-	collectorOnce.Do(func() {
-		collector = &MetricsCollector{
+	baseCollectorOnce.Do(func() {
```

Comment:
Why does this need to be in a `once.Do` if it's getting called from `init()`? Are there other places calling this to initialize it?

Reply:
ooo, I think I forgot to get rid of the sync.Once after Dustin recommended using init().